open-company/open-company-web
# Fix Sentry browser

NB: this is urgent only because right now in Sentry we get only the errors and the stack traces, no info about OS, browser and deploy release version.

Review with:
- [Infra](https://github.com/belucid/open-company-infrastructure/pull/130)

Sometimes Sentry doesn't add the OS and browser info.
Also `oc.web.lib.sentry/capture-error-with-context!` and `oc.web.lib.sentry/capture-error-with-message` are broken since they don't catch the error they create, so we get a report but w/o all the needed details and also it's not handled.


To test:
- open staging
- in console run `oc.web.lib.sentry.test_sentry()`
- do you see a message in Sentry? (Message from clojure)
- do you see also an error? (`ReferenceError: errorThrowingCode is not defined`)
- do you see release, user, OS and browser info on both?
- now run `oc.web.lib.sentry.capture_error_with_message("My first error", "My error message");`
- check the error contains release, user, OS and browser info
